### PR TITLE
Replace JodaTime library

### PR DESCRIPTION
**Pull request by JamesGoodwin**
- Use SimpleDateFormatter for parsing dates returned from partner API
- Use Calendar to replace JodaTime for card date expiry date/start date date calculations

### CODE REVIEW

File: judo-sdk/src/main/java/com/judopay/api/DateJsonDeserializer.java [hunk 1]
```diff
@@ -19,11 +20,30 @@ public Date deserialize(JsonElement json, Type typeOfT, JsonDeserializationConte
...
+     * @return the parsed date
+     * @throws ParseException
+     */
+    public Date iso8601ToDate(String date) throws ParseException {
+        try {
+            date = date.substring(0, 22) + date.substring(23);  // to remove the ":"
+        } catch (IndexOutOfBoundsException e) {
+            throw new ParseException("Invalid length", 0);
+        }
+        return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX", Locale.US).parse(date);
```
Comment (akakjs) [inline judo-sdk/src/main/java/com/judopay/api/DateJsonDeserializer.java]: So comparing with a sample "2016-03-22T11:14:29.3340+00:00" , it appears we use four decimal places for the fraction (no idea why...).

File: judo-sdk/src/test/java/com/judopay/api/DateJsonDeserializerTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        DateJsonDeserializer deserializer = new DateJsonDeserializer();
+
+        when(jsonElement.getAsString()).thenReturn("2016-03-24T09:55:28.3299+00:00");
+        Date date = deserializer.deserialize(jsonElement, type, jsonDeserializationContext);
+
+        assertThat(date.getTime(), is(1458813328329L));
+    }
+
+    @Test
+    public void shouldDeserializeDateWithTimezone() {
```
Comment (akakjs) [inline judo-sdk/src/test/java/com/judopay/api/DateJsonDeserializerTest.java]: I would test both positive and negative timezones

File: judo-sdk/src/main/java/com/judopay/api/DateJsonDeserializer.java [hunk 3]
```diff
@@ -19,11 +20,30 @@ public Date deserialize(JsonElement json, Type typeOfT, JsonDeserializationConte
...
+    /**
+     * Transforms an ISO-8601 String to a Date
+     *
+     * @param date the date as defined in ISO-8601 format
+     * @return the parsed date
+     * @throws ParseException
+     */
+    public Date iso8601ToDate(String date) throws ParseException {
+        try {
+            date = date.substring(0, 22) + date.substring(23);  // to remove the ":"
```
Comment (akakjs) [inline judo-sdk/src/main/java/com/judopay/api/DateJsonDeserializer.java]: As the number of decimal places in the fractional second could be variable, you might not want to hard code the location of the last ":" :P dare I say regex time? "[\+\-]\d\d\:\d\d"

**Comment by JamesGoodwin** [comment]
SimpleDateFormatter has some issues with handling timezone dates (reason for test failures).

Found an example of successful iso-8601 formatting in Android that's worth trying out:
https://gist.github.com/mraccola/702330625fad8eebe7d3

**Comment by Ryce** [comment]
:+1:

**Pull request merged**